src/plugin/plugin.ts

```diff
@@ -0,0 +1,320 @@
+      [productIdentifiers, type]
+    );
+  }
+
+  /**
+   * Success callback used when making a purchase. This callback will be triggered after a successful purchase.
+   * Cancelled purchases will trigger an error callback.
+   *
+   * @callback makePurchaseSuccessCallback
+   * @param {string} productIdentifier - The product identifier of the purchased product.
+   * @param {object} purchaserInfo - The updated Purchaser Info.
+   */
+
+  /**
+   * Error callback triggered when making a purchase. Cancelled purchases will also trigger this callback and
+   * error.userCancelled will be true
+   *
+   * @callback makePurchaseErrorCallback
+   * @param {object} error - The error containing message, code, domain and if user cancelled the purchase.
+   * @param {Boolean} userCancelled - Wether the user cancelled the purchase or not.
+   */
+
+  /**
+   * Make a purchase
+   * @param {string} productIdentifier The product identifier of the product you want to purchase.
+   * @param {makePurchaseSuccessCallback} callback Callback triggered after a successful purchase.
+   * @param {makePurchaseErrorCallback} errorcallback Callback triggered after an error or when the user cancels the purchase.
+   * If user cancelled, userCancelled will be true
+   * @param {String?} oldSku Optional sku you wish to upgrade from.
+   * @param {String} type Optional type of product, can be inapp or subs. Subs by default
+   */
+  public static makePurchase(
+    productIdentifier: string,
+    callback: any,
+    errorcallback: any,
+    oldSku: string | null,
+    type: string = "subs"
+  ) {
+    if (Array.isArray(oldSku)) {
+      throw new Error("Calling a deprecated method!");
+    }
+    window.cordova.exec(callback, errorcallback, PLUGIN_NAME, "makePurchase", [
+      productIdentifier,
+      oldSku,
+      type,
+    ]);
+  }
+
+  /**
+   * Restores a user's previous purchases and links their appUserIDs to any user's also using those purchases.
+   * @param {PurchaserInfoCallback} callback Callback that will receive the new purchaser info after restoring transactions.
+   * @param {ErrorCallback} errorcallback Callback that will be triggered whenever there is any problem restoring the user transactions. This gets normally triggered if there
+   * is an error retrieving the new purchaser info for the new user or the user cancelled the restore
+   */
+  public static restoreTransactions(callback: any, errorcallback: any) {
+    window.cordova.exec(
+      callback,
+      errorcallback,
+      PLUGIN_NAME,
+      "restoreTransactions",
+      []
+    );
+  }
+
+  /**
+   * Success callback used when calling a method that returns back a string
+   *
+   * @callback StringCallback
+   * @param {String} string - The returned String.
+   */
+
+  /**
+   * Get the appUserID that is currently in placed in the SDK
+   * @param {StringCallback} callback Callback that will receive the current appUserID
+   */
+  public static getAppUserID(callback: any) {
```

Comment:
I think this should be `callback: StringCallback`

Reply:
can we make types for all these callbacks?

Reply:
yeah I tried to do the minimum amount of changes with my context since I didn't want to introduce regressions where a type might error out in existing users setups.

Perhaps we can do this incrementally?

Reply:
yeah I agree, cool